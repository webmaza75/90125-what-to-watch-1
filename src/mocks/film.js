const film = {
  genre: [`Drama`, `Thriller`],
  title: `Aviator`,
  desc: ``,
  picture: `aviator.jpg`,
  year: 2016,
  src: ``
};

export default film;
