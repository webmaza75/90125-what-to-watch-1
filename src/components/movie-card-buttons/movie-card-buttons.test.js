import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MovieCardButtons from './movie-card-buttons.jsx';
import film from '../../mocks/film.js';

describe(`MovieCardButtons correctly renders`, () => {
  it(`MovieCardButtons correctly renders for authorized user`, () => {
    const renderer = new ShallowRenderer();
    const result = renderer
      .render(<MovieCardButtons
        isAuthorized={true}
        isFavorite={false}
        id={film.id}
        onMyListClick={jest.fn()}
        showAddReviewLink={false}
        isPromo={true}
      />);

    expect(result).toMatchSnapshot();
  });
});
