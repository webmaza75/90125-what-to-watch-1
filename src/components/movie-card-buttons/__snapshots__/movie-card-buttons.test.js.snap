// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardButtons correctly renders MovieCardButtons correctly renders for authorized user 1`] = `
<div
  className="movie-card__buttons"
>
  <button
    className="btn btn--play movie-card__button"
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </button>
  <Memo(ConnectFunction)
    id={1}
    isFavorite={false}
    isPromo={true}
  />
</div>
`;
